variable/util.cpp

```diff
@@ -47,11 +47,19 @@ Variable islinspace(const Variable &var, const Dim dim) {
                    "islinspace");
 }
 
+/// Return a variable of True, if variable values are sorted along given dim.
+///
+/// If `order` is SortOrder::Ascending, checks if values are non-decreasing.
+/// If `order` is SortOrder::Descending, checks if values are non-increasing.
+Variable issorted(const Variable &x, const Dim dim, const SortOrder order) {
+  return makeVariable<bool>(Values{allsorted(x, dim, order)});
+}
+
 /// Return true if variable values are sorted along given dim.
 ///
 /// If `order` is SortOrder::Ascending, checks if values are non-decreasing.
 /// If `order` is SortOrder::Descending, checks if values are non-increasing.
-bool issorted(const Variable &x, const Dim dim, const SortOrder order) {
+bool allsorted(const Variable &x, const Dim dim, const SortOrder order) {
```

Comment:
This is not quite what we want:
- We should change `allsorted` to rely on `issorted`.
- `issorted` should only "reduce" the requested dimension, rather than all dimensions, i.e., the output has dims of input, with `dim` removed.

One way to achieve this is as in `islinspace` above.

Reply:
> We should change allsorted to rely on issorted.
The reason it's presently the other way around is because it removes the need to create a Variable of bool, and then turn that into a bool again. It just makes a little more sense in terms of code flow this way round.

> issorted should only "reduce" the requested dimension, rather than all dimensions, i.e., the output has dims of input, with dim removed.

To clarify my understanding of the problem:
- `issorted` should actually perform the sorting, and not just check whether or not a variable in the given dim is sorted?
    - This understanding came from your use of the term "reduce" which I may be misinterpreting.
- `issorted` I think presently only checks the Dim that is passed in, could you point me at the area of code that doesn't do this? because I would like to fix it.

Reply:
No, what it should do is check whether data is sorting along a dimension. Example for a 2d variable as input:
```python
# Input:
# a b c
# a c b
out = sc.issorted(var, column);
# out:
# True
# False
```
The `dims=[row, column]` data is "reduced" to `dims=[column]`. Try yourself how this works, e.g., with `sc.max(var, dim)` to see the expected behavior.

Reply:
The output will always have one dimension fewer than the input, namely the dimension that is passed in is removed.

Reply:
So just to confirm my understanding for 3D data this is what we would expect:
var = sc.Variable(dims=['x', 'y', 'z'], values=[[[1,2,3], [1,2,3], [1,2,3]], [[1,2,3], [1,2,3], [1,2,3]], [[1,2,3], [1,2,3], [1,2,3]]])
# for issorted(var, 'z')
expected_var = sc.Variable(dims=['x', 'y'], values=[[True, True, True], [True, True, True], [True, True, True]])

Reply:
👍